scipy/sparse/linalg/dsolve/linsolve.py

```diff
@@ -79,7 +79,14 @@ def _get_umf_family(A):
             % (f_type, i_type)
         raise ValueError(msg)
 
-    return family
+    if np.prod(A.shape) > np.iinfo(np.int32).max:
+        family = family[0] + "l"
+        A_new = A.__class__(A.shape)
+        A_new.data = A.data
+        A_new.indptr = np.array(A.indptr, copy=False, dtype=np.int64)
+        A_new.indices = np.array(A.indices, copy=False, dtype=np.int64)
+
+    return family, A_new
```

Comment:
You need to add an `else` block that assigns `A_new = A`.

Reply:
🤦x2